Lib/test/test_zoneinfo/test_zoneinfo.py

```diff
@@ -1717,11 +1718,31 @@ def test_env_variable_relative_paths(self):
         for input_paths, expected_paths in test_cases:
             path_var = os.pathsep.join(input_paths)
             with self.python_tzpath_context(path_var):
+                with self.subTest("module-level warning", path_var=path_var):
+                    with CleanImport("zoneinfo", "zoneinfo._tzpath"):
+                        with warnings.catch_warnings(record=True) as wlog:
```

Comment:
Why not use `assertWarns()`?

Reply:
You can use `assertWarns(RuntimeWarning)` and then check the type of the exception more precisely. Also you can use the reference to `zoneinfo.InvalidTZPathWarning` saved in the `CleanImport` block.

And since the main purpose of this check is to check that stacklevel is correct, I recommend to split it in other test method and only check with a single invalid path.

Reply:
I moved it to a new test function with just a single invalid file + I am now using `assertWarns` instead of `catch_warnings` 👍